src/hooks/useParaswapCallback.ts

```diff
@@ -117,25 +117,22 @@ export function useParaswapCallback(
         summary: string;
       }> {
         const pct = basisPointsToPercent(allowedSlippage);
-        const srcAmount = trade
-          .maximumAmountIn(pct)
-          .multiply(JSBI.BigInt(10 ** trade.inputAmount.currency.decimals))
-          .toFixed(0);
+
+        const srcAmount =
+          priceRoute.side === SwapSide.SELL
+            ? priceRoute.srcAmount
+            : priceRoute.destAmount;
         const minDestAmount = trade
           .minimumAmountOut(pct)
           .multiply(JSBI.BigInt(10 ** trade.outputAmount.currency.decimals));
 
         const referrer = 'Quickswap';
 
-        const srcToken = getBestTradeCurrencyAddress(
-          trade.inputAmount.currency,
-        );
-        const destToken = getBestTradeCurrencyAddress(
-          trade.outputAmount.currency,
-        );
+        const srcToken = priceRoute.srcToken;
+        const destToken = priceRoute.destToken;
 
-        const srcDecimals = trade.inputAmount.currency.decimals;
-        const destDecimals = trade.outputAmount.currency.decimals;
+        const srcDecimals = priceRoute.srcDecimals;
+        const destDecimals = priceRoute.destDecimals;
 
         //Update the rate before calling swap
         const rate = await paraswap.getRate({
```

Comment:
We may want to consider removing the price update here since the rate is now updated every second in the swap component.

Reply:
I think we need to use same parameters of the rate objects passed from the swap component